csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -121,7 +125,7 @@ public override void Dispose()
                 this.client.CloseSession(r6).Wait();
 
                 this.transport?.Close();
-                this.client.Dispose();
+                if (this.client is IDisposable disposable) disposable.Dispose();
```

Comment:
Need to be more generic here because we're not sure it implements `IDisposable`.